core/server/worker/src/test/java/alluxio/worker/block/DefaultBlockWorkerTest.java

```diff
@@ -297,4 +445,208 @@ public void getBlockReader() throws Exception {
+        .newBuilder()
+        .setUfsPath("/nonexistent/path")
+        .setBlockInUfsTier(true)
+        .setBlockSize(1024)
+        .setMountId(UFS_MOUNT_ID)
+        .build();
+
+    assertThrows(UnavailableException.class,
+        () -> mBlockWorker.createUfsBlockReader(
+            sessionId, blockId, 0, false, options));
+  }
+
+  @Test
+  public void cacheBlockSync() throws Exception {
+    cacheBlock(false);
+  }
+
+  @Test
+  public void cacheBlockAsync() throws Exception {
+    cacheBlock(true);
+  }
+
+  @Test
+  public void getConfiguration() {
+    alluxio.wire.Configuration conf = mBlockWorker.getConfiguration(
+        GetConfigurationPOptions
+            .newBuilder()
+            .setIgnoreClusterConf(false)
+            .setIgnorePathConf(false)
+            .setRawValue(true)
+            .build());
+    assertEquals(conf.getClusterConfHash(), Configuration.hash());
+  }
+
+  @Test
+  public void cleanUpSession() throws Exception {
+    long sessionId = mRandom.nextLong();
+    long blockId = mRandom.nextLong();
+
+    mBlockWorker.createBlock(
+        sessionId,
+        blockId,
+        0,
+        new CreateBlockOptions(null, Constants.MEDIUM_MEM, 1));
+    mBlockWorker.commitBlock(sessionId, blockId, false);
+
+    // just to hold a read lock on the block
+    BlockReader reader = mBlockWorker.createBlockReader(sessionId, blockId, 0, false, null);
+
+    long anotherSessionId = mRandom.nextLong();
+
+    // clean up the first session
+    mBlockWorker.cleanupSession(sessionId);
+
+    // now another session should be able to grab write lock on the block
+    mBlockWorker.removeBlock(anotherSessionId, blockId);
+  }
+
+  private void cacheBlock(boolean async) throws Exception {
+    // flush 1MB random data to ufs so that caching will take a while
+    long ufsBlockSize = 1024 * 1024;
+    byte[] data = new byte[(int) ufsBlockSize];
+    mRandom.nextBytes(data);
+
+    try (FileOutputStream fileOut = new FileOutputStream(mTestUfsFile);
+         BufferedOutputStream bufOut = new BufferedOutputStream(fileOut)) {
+      bufOut.write(data);
+      bufOut.flush();
+    }
+
+    // ufs options: delegate to the ufs mounted at UFS_MOUNT_ID
+    // with path to our test file
+    long blockId = mRandom.nextLong();
+    Protocol.OpenUfsBlockOptions options = Protocol.OpenUfsBlockOptions
+        .newBuilder()
+        .setBlockSize(ufsBlockSize)
+        .setUfsPath(mTestUfsFile.getAbsolutePath())
+        .setMountId(UFS_MOUNT_ID)
+        .setNoCache(false)
+        .setOffsetInFile(0)
+        .build();
+
+    // cache request:
+    // delegate to local ufs client rather than remote worker
+    CacheRequest request = CacheRequest
+        .newBuilder()
+        .setSourceHost(NetworkAddressUtils.getLocalHostName(500))
+        .setBlockId(blockId)
+        .setLength(ufsBlockSize)
+        .setAsync(async)
+        .setOpenUfsBlockOptions(options)
+        .build();
+
```

Comment:
Interestingly the caching doesn't work if length is not set in CacheRequest. I wonder if this information is just a replicate of OpenUfsBlockOptions#block_size.